dask_glm/utils.py

```diff
@@ -8,6 +8,7 @@
 from multipledispatch import dispatch
 
 
+@dispatch(np.ndarray)
```

Comment:
I don't think that this is necessary.  I also think that this is causing some of your errors.

This function should work on all objects the same.  The different array classes are handled by dispatching on `exp` within this function.